telegram/files/file.py

```diff
@@ -131,52 +130,45 @@ def download(
         if custom_path is not None and out is not None:
             raise ValueError('custom_path and out are mutually exclusive')
 
-        local_file = is_local_file(self.file_path)
-
-        if local_file:
-            url = self.file_path
-        else:
-            # Convert any UTF-8 char into a url encoded ASCII string.
-            url = self._get_encoded_url()
+        local_file: bool = is_local_file(self.file_path)
+        url: Union[Path, str] = Path(self.file_path) if local_file else self._get_encoded_url()
 
         if out:
             if local_file:
-                with open(url, 'rb') as file:
-                    buf = file.read()
+                buf = url.read_bytes()  # type: ignore[union-attr]
             else:
-                buf = self.bot.request.retrieve(url)
+                buf = self.bot.request.retrieve(url)  # type: ignore[arg-type]
                 if self._credentials:
                     buf = decrypt(
                         b64decode(self._credentials.secret), b64decode(self._credentials.hash), buf
                     )
             out.write(buf)
             return out
 
-        if custom_path and local_file:
-            shutil.copyfile(self.file_path, custom_path)
-            return custom_path
+        if custom_path is not None and local_file:
+            shutil.copyfile(self.file_path, str(custom_path))
+            return Path(custom_path)
 
         if custom_path:
-            filename = custom_path
+            filename = Path(custom_path)
         elif local_file:
-            return self.file_path
+            return Path(self.file_path)
         elif self.file_path:
-            filename = basename(self.file_path)
+            filename = Path(Path(self.file_path).name)
```

Comment:
I think we can just return `Path(self.file_path)`. having more info in the return value is less critical than having less info. IISC the use of `basename` was introduced 6 years ago, so it's probably not worth it to try & recreate why it was used back then …

Reply:
no problem, will do, i was afraid to break compatibility with usages of this variable.

Reply:
our credo for v14 is to break *all* the things 🙆🏼‍♂️🥳 just make sure to add a versioning directive for it :)

Reply:
done

Reply:
reverted.
Reason - ```TestFile.test_download()``` failed: 
```
OSError: [Errno 22] Invalid argument: 'https:\\api.org\\file\\bot133505823:AAHZFMHno3mzVLErU5b5jJvaeG--qUyLyG0\\document\\file_3'
```

The error indicates a little something i had forgotten - the ```custom_path``` argument is not necessarily a path, it can also be a URL, (and maybe even a file URI?), and in such cases, the ```basename``` is, if i understand the error correctly, most reuired, rather than the entire url (```file_3``` in the error presented above)

Reply:
Ah, yes this makes sense 👍